src/Android/Xamarin.Android/Samples/Layers/QueryCQLFilters/QueryCQLFilters.cs

```diff
@@ -0,0 +1,354 @@
+            _applyButton = FindViewById<Button>(Resource.Id.applyButton);
+            _resetButton = FindViewById<Button>(Resource.Id.resetButton);
+
+            _progressBar = FindViewById<ProgressBar>(Resource.Id.progressBar);
+
+            _dateSwitch.CheckedChange += DateSwitchChanged;
+
+            _startDateButton.Click += StartDateClick;
+            _endDateButton.Click += EndDateClick;
+
+            _applyButton.Click += ApplyQuery;
+            _resetButton.Click += ResetQuery;
+        }
+
+        private void ResetQuery(object sender, EventArgs e)
+        {
+            _whereClauseSpinner.SetSelection(0);
+            _maxFeatures.Text = "3000";
+            _dateSwitch.Checked = true;
+            _startTime = new DateTime(2011, 6, 13);
+            _endTime = new DateTime(2012, 1, 7);
+            _startDateButton.Text = ((DateTime)_startTime).ToShortDateString();
+            _endDateButton.Text = ((DateTime)_endTime).ToShortDateString();
+        }
+
+        private async void ApplyQuery(object sender, EventArgs e)
+        {
+            if (_featureTable.LoadStatus != Esri.ArcGISRuntime.LoadStatus.Loaded)
+            {
+                return;
+            }
+
+            try
+            {
+                _progressBar.Visibility = Android.Views.ViewStates.Visible;
+
+                // Set queryParameters to the user's input
+                var queryParameters = CreateQueryParameters();
+
+                // Populate the table with the query, clearing the exist content of the table.
+                // Setting outFields to null requests all fields.
+                var result = await _featureTable.PopulateFromServiceAsync(queryParameters, true, outFields: null);
+
+                // Zoom to the extent of the returned features.
+                Envelope tableExtent = GeometryEngine.CombineExtents(result.Select(feature => feature.Geometry));
+                if (tableExtent != null && !tableExtent.IsEmpty)
+                {
+                    await _myMapView.SetViewpointGeometryAsync(tableExtent, 20);
+                }
+
+                // Report the number of returned features by the query.
+                new AlertDialog.Builder(this).SetMessage($"Query returned {result.Count()} features.").SetTitle("Query Completed").Show();
+            }
+            catch (Exception ex)
+            {
+                new AlertDialog.Builder(this).SetMessage(ex.Message).SetTitle(ex.GetType().Name).Show();
+            }
+            finally
+            {
+                // Update the UI.
+                _progressBar.Visibility = Android.Views.ViewStates.Gone;
+            }
+        }
+
+        private async void StartDateClick(object sender, EventArgs e)
+        {
+            try
+            {
+                DateTime promptDate = new DateTime(2011, 6, 13);
+                if (_startTime is DateTime validStart) promptDate = validStart;
```

Comment:
I recommend using braces here. Also, I recommend adding comments to the next two methods.